### PR TITLE
Ensuring data-scientists group is added in prod

### PR DESCRIPTION
**Pull request by damoodamoo**
DS AAD group needs to exist in prod.

### CODE REVIEW

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4736134829 (with refid `8342a5b0`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/262#issuecomment-1513675631) from @damoodamoo)

File: infrastructure/core/locals.tf [hunk 1]
```diff
@@ -31,5 +31,6 @@ locals {
     cosmos     = "privatelink.documents.azure.com"
     databricks = "privatelink.azuredatabricks.net"
     sql        = "privatelink.database.windows.net"
+    postgres   = "privatelink.postgres.database.azure.com"
```
Comment (jjgriff93) [inline infrastructure/core/locals.tf:34]:
```suggestion
```
As per the slack thread, this shouldn't be added here

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4736825797 (with refid `8342a5b0`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/262#issuecomment-1513775458) from @damoodamoo)

File: infrastructure/transform/locals.tf [hunk 2]
```diff
@@ -96,7 +96,8 @@ locals {
   msi_users = [
     { "name" : "${local.databricks_app_name}", "role" : "db_owner" },
     { "name" : "${azuread_group.ad_group_apps.display_name}", "role" : "db_datareader" },
+    local.data_scientists
```
Comment (jjgriff93) [inline infrastructure/transform/locals.tf:99]: Don't think this should be in the msi_users? MSI is for system identities right? Why did you move this from the concat below?

**Review by t-young31 — APPROVED** [review]
LGTM

File: infrastructure/transform/locals.tf [hunk 3]
```diff
@@ -96,7 +96,8 @@ locals {
...
   ]
 
-  sql_users_to_create = var.accesses_real_data ? local.msi_users : concat(local.msi_users, [local.developers, local.data_scientists])
+  sql_users_to_create = var.accesses_real_data ? local.msi_users : concat(local.msi_users, [local.developers])
```
Comment (jjgriff93) [inline infrastructure/transform/locals.tf:102]: I would just have a concat of `local.msi_users` & data scientists on the left hand side for this to be more explicit

Comment (damoodamoo) [reply infrastructure/transform/locals.tf]: yep - made it way clearer now :)

File: infrastructure/transform/locals.tf (hunk 2 above)
Comment (damoodamoo) [reply infrastructure/transform/locals.tf]: as below

**Review by jjgriff93 — APPROVED** [review]
Love it

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4741459256 (with refid `8342a5b0`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/262#issuecomment-1514379492) from @jjgriff93)

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4741570505 (with refid `8342a5b0`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/262#issuecomment-1514395128) from @damoodamoo)

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

`/test-destroy-env` is not recognised as a valid command.

You can use the following commands:
&nbsp;&nbsp;&nbsp;&nbsp;/test - deploy any modified modules and destroy
&nbsp;&nbsp;&nbsp;&nbsp;/test-all - deploy everything (ignoring path filters) and destroy
&nbsp;&nbsp;&nbsp;&nbsp;/test-force-approve - force approval of the PR tests (i.e. skip the deployment checks)
&nbsp;&nbsp;&nbsp;&nbsp;/destroy - delete the PR environment (e.g. to enable testing a deployment from a clean start after previous tests)
&nbsp;&nbsp;&nbsp;&nbsp;/destroy-no-terraform - delete the PR environment without terraform
&nbsp;&nbsp;&nbsp;&nbsp;/help - show this help

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/262#issuecomment-1514422673) from @damoodamoo)

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

Destroying environment (with refid 8342a5b0)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/262#issuecomment-1514423305) from @damoodamoo)

**Comment by damoodamoo** [comment]
/test-force-approve

build + destroy failed on AML storage account, unrelated

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:white_check_mark: Marking tests as complete (for commit c16fb68e717ecc4ab20e158c5c66e120c61b23be)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/262#issuecomment-1514437120) from @damoodamoo)
